crates/circuit/src/variable_mapper.rs

```diff
@@ -0,0 +1,267 @@
+            Ok(Target::Register(register))
+        } else {
+            Ok(Target::Expr(ob.extract()?))
+        }
+    }
+}
+
+pub(crate) struct VariableMapper {
+    target_cregs: Vec<ClassicalRegister>,
+    register_map: RefCell<HashMap<String, ClassicalRegister>>,
+    bit_map: HashMap<ShareableClbit, ShareableClbit>,
+    var_map: HashMap<expr::Var, expr::Var>,
+    stretch_map: HashMap<expr::Stretch, expr::Stretch>,
+}
+
+impl VariableMapper {
+    /// Constructs a new mapper.
+    ///
+    /// The `stretch_map` is only used for direct calls to [VariableMapper::map_expr]
+    /// since `condition`s and `target`s expressions are never durations. Provide
+    /// [None] if you don't need this.
+    pub fn new(
+        target_cregs: Vec<ClassicalRegister>,
+        bit_map: HashMap<ShareableClbit, ShareableClbit>,
+        var_map: HashMap<expr::Var, expr::Var>,
+        stretch_map: Option<HashMap<expr::Stretch, expr::Stretch>>,
+    ) -> Self {
+        Self {
+            target_cregs,
+            register_map: RefCell::default(),
+            bit_map,
+            var_map,
+            stretch_map: stretch_map.unwrap_or_default(),
+        }
+    }
+
+    /// Map the given `condition` so that it only references variables in the destination
+    /// circuit (as given to this class on initialization).
+    ///
+    /// If `allow_reorder` is `true`, then when a legacy condition (the two-tuple form) is made
+    /// on a register that has a counterpart in the destination with all the same (mapped) bits but
+    /// in a different order, then that register will be used and the value suitably modified to
+    /// make the equality condition work.  This is maintaining legacy (tested) behavior of
+    /// [DAGCircuit::compose]; nowhere else does this, and in general this would require *far*
+    /// more complex classical rewriting than Qiskit needs to worry about in the full expression
+    /// era.
+    pub fn map_condition<F>(
+        &self,
+        condition: &Condition,
+        allow_reorder: bool,
+        mut add_register: F,
+    ) -> PyResult<Condition>
+    where
+        F: FnMut(&ClassicalRegister) -> PyResult<()>,
+    {
+        Ok(match condition {
+            Condition::Bit(target, value) => {
+                Condition::Bit(self.bit_map.get(target).unwrap().clone(), *value)
+            }
+            Condition::Expr(e) => Condition::Expr(self.map_expr(e, &mut add_register)?),
+            Condition::Register(target, value) => {
+                if !allow_reorder {
+                    return Ok(Condition::Register(
+                        self.map_register(target, &mut add_register)?,
+                        *value,
+                    ));
+                }
+                // This is maintaining the legacy behavior of `DAGCircuit.compose`.  We don't
+                // attempt to speed-up this lookup with a cache, since that would just make the more
+                // standard cases more annoying to deal with.
```

Comment:
```suggestion
                }
                // This is maintaining the legacy behavior of `DAGCircuit.compose`.  We don't
                // attempt to speed-up this lookup with a cache, since that would just make the
                // standard cases more annoying to deal with.
```
Saying "more" twice here seems a bit redundant.

Reply:
I will defer this since the comment is just ported from the Python version 🙂